### DISCUSSION TITLE
Water valve automatic ios

### DISCUSSION THREAD

**Discussion by Dimbo6022** [RavenSystem/esp-homekit-devices#1741] [MEPLHAA Scripts Collection]
good afternoon. I ask for help in writing json in order for automation to work in the native home application on ios.
i used the following code:
{
  "c": { "l": 0, "b": [{ "g": 2, "t": 5 }] },
  "a": [{
    "t": 20,
    "0": { "r": [{ "g": 15, "v": 1 }, { "g": 4 }] },
    "1": { "r": [{ "g": 15 }, { "g": 4, "v": 1 }] },    
    "s": 0,
    "w": 3,
    "d": 0
  }]
}
I'm sorry I'm still new to this skill.

**Comment by Dimbo6022** [comment]
Если ещё и по русски получится объяснить будет 10 из 10.

**Comment by Ucoz3000** [comment]
{
  "c": {"l": 0,"b": [{"g": 2,"t": 5}]},
  "a": [{
      "0": {"r": [{"g": 15,"v": 0}]},
      "1": {"r": [{"g": 15,"v": 1}]},
      "t": 20,
      "b": [{ "g": 0 }],
      "s": 0,
      "w": 3,
      "d": 3600
    }]
}


попробуй

**Comment by Ucoz3000** [comment]
"l": 0            <- Enable status LED at  GPIO 0
"b": [           <- Buttons. Must be an array
"g": 2,         <- First button at GPIO 2
"t": 5           <- Set button to single press type
"a": [          <- Accessory section. Must be an array
"0": {          <- Actions triggered when HomeKit switch is set to OFF
"r": [           <- Digital outputs. Must be an array
"g": 15,       <- Relay connected to GPIO 15
"v": 0          <- Value assigned to the relay 
"1": {          <- Actions triggered when HomeKit switch is set to ON
"r": [           <- Digital outputs. Must be an array
"g": 15,       <- Relay connected to GPIO 15
"v": 1          <- Value assigned to the relay
"t": 20,       <- Accessory type 1 (Water Value)
"b": [          <- Buttons. Must be an array
"g": 0,        <- First button at GPIO 0
"s": 0,         <- State a switch service enters on boot
"w": 3,         <- Valve Type (Tap / Faucet)
"d": 3600    <- Timer set to 1 hour (default)

**Reply by Dimbo6022** [reply]
Попробовал, автоматизации нет. нашёл код для жалюзей. Автоматизация есть.
 
{
  "c": { "l": 0, "b": [{ "g": 2, "p": 0, "t": 5 }] },
  "a": [{
    "t": 45,
    "o": 10,
    "c": 10,
    "f": 10,
    "0": { "r": [{ "g": 4, "v": 1 }, { "g": 15 }] },
    "1": { "r": [{ "g": 4 }, { "g": 15, "v": 1 }] },
    "2": { "r": [{ "g": 4 }, { "g": 15 }] },
    "3": { "r": [{ "g": 4, "i": 1 }, { "g": 15 }] },
    "4": { "r": [{ "g": 4 }, { "g": 15, "i": 1 }] },
    "f0": [{ "g": 5, "p": 0 }],
    "f1": [{ "g": 13, "p": 0 }],
    "f2": [{ "g": 5, "t": 0 }, { "g": 13, "t": 0 }]
  }]
}

пробовал заменить t с жалюзей на кран - автоматизации нет.

**Comment by xrust83** [comment]
Homekit не підтримує і ніколи не підтримував автоматизації пов’язані з водою і кранами. Різниця від звичайної кнопки це вимкнення за встановленим часом, щоб не забути. Якщо хочете вмикати від автоматизації: 1 - ставите звичайний вимикач в HomeKit, 2 - робите автоматизацію в HomeKit на будь який вимикач, а потім через  сторонню програму, наприклад Controller для Homekit чи Home+, робите підміну замінивши вимикач краном. 
3. Через додаток EVE робите сценарій, який вмикає і вимикає кран, а тоді через HomeKit робите автоматизацію, яка вмикає и вимикає сценарій. Якщо потрібно змінити налаштування з часом теж через сторонню програму. Раніше через такі милиці автоматизації з температурою налаштовувалися, заказ вже температуру як тригер можна використовувати з Homekit, а крани ще не можна.

**Reply by Dimbo6022** [reply]
Спасибо за информацию. Думаю тогда останусь с рабочим сценарием когда кран выглядит в приложении Дом как жалюзи. Но зато с автоматизацией.

**Reply by xrust83** [reply]
Через підміну він виглядає і працює як кран, також через підміну вимикається від датчиків затоплення. 

![C0183426-3A57-451D-8C9E-169AD959CC16](https://user-images.githubusercontent.com/58448955/154809840-5a067c90-690c-4c6a-b4e5-2dad113fc835.png)
![C661C5D9-31A7-4FA3-A3FF-F37A93E03817](https://user-images.githubusercontent.com/58448955/154809849-f088e02a-3e07-4cfc-8f51-cd0ec410ed82.png)
![6DE84DE0-CAC7-41EE-91A1-27D4B0FF3D44](https://user-images.githubusercontent.com/58448955/154810328-e80d7c27-8316-4cfc-a407-71d4133ca07f.png)

**Comment by VaBanck** [comment]
https://discord.gg/v8hyxj2 заходите, тут есть русский чат

**Reply by xrust83** [reply]
Я читаю :-). Дякую
